DNN Platform/DotNetNuke.Abstractions/Settings/ISaveSettingsService.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+namespace DotNetNuke.Abstractions.Settings
+{
+    /// <summary>
+    /// The Save Settings service is used by implementations of
+    /// <see cref="ISettingsService"/>. It is not currently designed
+    /// to be injected via Dependency Injection. Please use the
+    /// <see cref="ISettingsService"/>.
+    /// </summary>
+    public interface ISaveSettingsService
```

Comment:
Should these interfaces be `internal` if they're not intended to be used via DI?

Reply:
This is a really great question!

In my opinion these interfaces should remain public if we want to support making the `PortalSettingsService` extensible. If a developer wants to swap it out for their own implementation they should be able to use the following code.

```c#
public class CustomPortalSettingsManager : PortalSettingsManager { }
public class CustomPortalSaveSettingsService : PortalSaveSettingsService { }
public class CustomPortalDeleteSettingsService : PortalDeleteSettingsService { }
```
If the extension is just changing the save/delete rules it doesn't necessarily need to create another class for the `PortalSettingsService`.This approach gives us the highest degree of extensibility, but also makes it more complicated.

For many modules and implementations this will never be used. I see this being valuable for something like Evoq or custom installations that want to execute special logic during the save or delete sequence.